.github/workflows/conda-cpp-tests.yaml

```diff
@@ -2,8 +2,14 @@ on:
   workflow_call:
     inputs:
       build_type:
+        description: "One of: [branch, nightly, pull-request]"
```

Comment:
This and other test workflows only get invoked after a pull-request build or nightly build, not after a branch build.

This decision is deliberate to avoid straining our GPU capacity.

Because of this, I'm not sure including a `branch` build type here makes sense since it will never get used.

Reply:
Making this possible is actually one of the goals here. See https://github.com/rapidsai/gha-tools/issues/138.

Reply:
To be clear, I'm not proposing something **new** here.

Currently on `branch-25.04`, there's already handling of `build_type: branch` in this workflow and the others touched by this PR.

https://github.com/rapidsai/shared-workflows/blob/9ebdde271fab64e3752b58b5b9878d804f69b6af/.github/workflows/conda-cpp-tests.yaml#L88-L93

This change just documents the current state.

I'm happy to try to make this stricter as part of this PR if you think there aren't any uses out there that trigger tests for `branch` builds... given what I just wrote, do you want me to do that?

Reply:
Wait actually, looking through the blame... the code I linked was just added around a month ago, by @bdice : #276

I think to support manually triggering a `test` job via `workflow_dispatch`.

Reply:
Whoops. I see that now. I misinterpreted the diff on your PR.

I think my comment can be disregarded than.